README.md

```diff
@@ -1,2 +1,107 @@
-# pystac-io
-Python library providing additional IO implementations for PySTAC
+# PySTAC IO
+
+PySTAC IO is a Python library that provides additional IO implementations for [`pystac.STAC_IO`](https://pystac.readthedocs.io/en/latest/concepts.html#using-stac-io).
+
+PySTAC supports reading and writing STAC catalogs via the local filesytem. The project provides:
+
+| Module | Scheme | Reads? | Writes? |
+|--------|------------|-------|--------|
+| https  | `https://` | X     |        |
+| s3     | `s3://`    | X     | X      |
+
+## Installation
+
+Clone then install locally into your environment via pip:
+
+```shell
+pip install .
+```
+
+You'll also need to install the dependencies for the modules you wish to use.
+For example, to use the `s3` module:  
+
+```shell
+pip install ".[s3]"
+```
+
+## Usage
+
+Import `pystac_io` and the submodule you plan to use, register the submodule to enable it, and then use `pystac` normally!
+
+```python
+import pystac
+import pystac_io
+import pystac_io.s3   # Import the IO module you plan to `register()`
+
+pystac_io.register("s3")
```

Comment:
The primary alternative I considered here was to just expose all of the underlying `{module}_[read|write]_text_method` functions and it's up to the user to manually apply them to `pystac.STAC_IO`. That would look something like:
```
import pystac
from pystac_io.s3 import s3_read_text_method, s3_write_text_method

pystac.STAC_IO.read_text_method = s3_read_text_method
pystac.STAC_IO.write_text_method = s3_write_text_method

s3_catalog = pystac.Catalog.from_file("s3://bucket/path/to/catalog.json")

# Revert to defaults
pystac.STAC_IO.read_text_method = pystac.STAC_IO.default_read_text_method
pystac.STAC_IO.write_text_method = pystac.STAC_IO.default_write_text_method
```

If you just want a read method and don't need to swap to different modules, its about the same code either way. If you need read+write and are swapping modules, the extra wrapper methods implemented in this PR reduce boilerplate. They also provide a simple interface that only requires one extra line of code for users to define and use their own modules. 

I implemented the API in this PR because it was the "harder" option and seemed useful to demonstrate what it ended up looking like. If it's too complicated for what's essentially a lose collection of methods and their dependencies I don't have any major concern with reverting to the version described here without the wrapper API.

Reply:
I'd lean towards a wrapper API that implements a context manager in light of https://github.com/azavea/pystac-io/pull/1#discussion_r513740683